--- smb.conf
+++ smb-ps2smb.conf.plus
@@ -346,3 +346,25 @@
 force user = root
 veto files = /._*/.DS_Store/
 delete veto files = yes
+
+[ps2smb]
+comment = Open PS2 Loader
+path = /userdata/roms/ps2smb
+writeable = yes
+guest ok = yes
+create mask = 0744
+directory mask = 0755
+force user = root
+veto files = /._*/.DS_Store/
+delete veto files = yes
+
+[storage]
+comment = Batocera.PLUS storage system
+path = /var/storage
+writeable = yes
+guest ok = yes
+create mask = 0744
+directory mask = 0755
+force user = root
+veto files = /._*/.DS_Store/
+delete veto files = yes
