--- configgen-defaults.yml
+++ configgen-defaults.yml.plus
@@ -77,7 +77,7 @@
   core:     genesisplusgx
 saturn:
   emulator: libretro
-  core:     yabasanshiro
+  core:     beetle-saturn
 dreamcast:
   emulator: libretro
   core: flycast
@@ -91,11 +91,11 @@
   emulator: libretro
   core:     fbneo
 neogeocd:
-  emulator: libretro
-  core:     neocd
+  emulator: raine
+  core:     raine2020
 mame:
   emulator: libretro
-  core:     mame078plus
+  core:     mame0200
 fbneo:
   emulator: libretro
   core:     fbneo
@@ -142,7 +142,7 @@
   core:     '81'
 dos:
   emulator: libretro
-  core: dosbox
+  core: dosbox_pure
   options:
     videomode: default
 c64:
@@ -252,11 +252,11 @@
   emulator: moonlight
   core: moonlight
 psp:
-  emulator: ppsspp
+  emulator: libretro
   core: ppsspp
 satellaview:
   emulator: libretro
-  core:     snes9x
+  core:     bsnes
 sufami:
   emulator: libretro
   core:     snes9x
@@ -277,7 +277,7 @@
   core:     np2kai
 lightgun:
   emulator: libretro
-  core:     mame078plus
+  core:     mame0200
 daphne:
   emulator: daphne
   core:     hypseus
@@ -291,7 +291,7 @@
     ratio: 0
     filter: 0
 windows:
-  emulator: wine
+  emulator: wine-staging
   core:     wine
 windows_installers:
   emulator: wine
@@ -302,3 +302,87 @@
 ps3:
   emulator: rpcs3
   core:     rpcs3
+n64dd:
+  emulator: libretro
+  core:     parallel_n64
+fba_libretro:
+  emulator: libretro
+  core:     fbalpha2012
+nesmini:
+  emulator: libretro
+  core:     fceumm
+snesmini:
+  emulator: libretro
+  core:     snes9x
+snescd:
+  emulator: libretro
+  core:     snes9x
+pspminis:
+  emulator: libretro
+  core:     ppsspp
+psxmini:
+  emulator: libretro
+  core:     pcsx_rearmed
+neogeomini:
+  emulator: libretro
+  core:     fbneo
+neogeox:
+  emulator: libretro
+  core:     fbneo
+channelf:
+  emulator: libretro
+  core:     freechaf
+hbmame:
+  emulator: libretro
+  core:     hbmame
+internet:
+  emulator: internet
+  core:     firefox
+flash:
+  emulator: internet
+  core:     firefox
+arcade:
+  emulator: libretro
+  core:     mame0200
+mugen:
+  emulator: wine
+  core:     wine
+model2:
+  emulator: model2
+  core:     model2
+megadrivemsu1:
+  emulator: libretro
+  core:     genesisplusgx
+tic80:
+  emulator: libretro
+  core:     tic80
+pico8:
+  emulator: libretro
+  core:     retro8
+yuzu:
+  emulator: yuzu
+  core:     yuzu
+x1:
+  emulator: libretro
+  core:     x1
+scv:
+  emulator: libretro
+  core:     emuscv
+uzebox:
+  emulator: libretro
+  core:     uzem
+palmos:
+  emulator: libretro
+  core:     mu
+samcoupe:
+  emulator: libretro
+  core:     simcp
+j2me:
+  emulator: j2me
+  core:     j2me
+vpinball:
+  emulator: vpinball
+  core:     vpinball
+fpinball:
+  emulator: fpinball
+  core:     fpinball
\ No newline at end of file
