diff --git a/ext/vulkan/vkerror.c b/ext/vulkan/vkerror.c
index 3fec27e..3dd14dd 100644
--- a/ext/vulkan/vkerror.c
+++ b/ext/vulkan/vkerror.c
@@ -63,8 +63,6 @@ _vk_result_to_string (VkResult result)
 
   if (result >= 0)
     return NULL;
-  if (result < VK_RESULT_BEGIN_RANGE)
-    return "Unknown Error";
 
   for (i = 0; i < G_N_ELEMENTS (vk_result_string_map); i++) {
     if (result == vk_result_string_map[i].result)
