--- batocera-save-overlay.original
+++ batocera-save-overlay.modificado
###
### Aumenta o tamanho padrão do arquivo overlay.
###
@@ -6,7 +6,7 @@
 OVERLAYFILE="/boot/boot/overlay"
 OVERLAYMOUNT="/overlay/saved"
 OVERLAYRAM="/overlay/overlay"
-OVERLAYSIZE=50 # M
+OVERLAYSIZE=256 # M
 
 # fix for winscp running bash and then not sourcing /etc/profile
 PATH=$PATH:/sbin

###
### Diminui o tempo para salvar as alterações.
###
@@ -18,7 +18,7 @@
 
     # X MB as ext4
     echo "Creating an overlay file on the /boot filesystem..."
-    dd if=/dev/zero of="${OVERLAYFILE}" bs=${COISIZE}M count=1 || return 1
+    dd if=/dev/zero of="${OVERLAYFILE}" bs=1 count=0 seek=${COISIZE}M || return 1
     echo "Formating the overlay file in ext4..."
     mkfs.ext4 "${OVERLAYFILE}"                         || return 1
 }

###
### Desativa a montagem da unidade como somente leitura se estiver rodando pelo batocera-hd-dition.
###
@@ -74,10 +74,13 @@
 fi
 
 # put /boot in ro back
-echo "Making /boot read only..."
-if ! mount -o remount,ro /boot
+if ! df | grep ^/dev/loop0 | grep /boot$ > /dev/null
 then
-    exit 1
+    echo "Making /boot read only..."
+    if ! mount -o remount,ro /boot
+    then
+        exit 1
+    fi
 fi
 
 echo "Synchronizing..."
