--- batocera-makepkg.original
+++ batocera-makepkg.modificado
@@ -107,7 +107,7 @@
 	pk=../"$1".pkg.tar.xz
 	[[ -f "$pk" ]] && echo "ERROR: file $pk already exists. Aborting." && exit 1
 	echo "Creating package $pk ..."
-	tar -cf - .PKGINFO * | xz -c -z - > "$pk"
+	tar -cf - .PKGINFO * | xz -c -z -9evT0 - > "$pk"
 	ret=$?
 	[[ $ret -eq 0 ]] && echo "SUCCESS: package $pk correctly generated" && exit 0
 }
