--- batocera-store.original
+++ batocera-store.modificado
@@ -1,5 +1,24 @@
 #!/bin/bash
 
+PACMAN_DIR=/userdata/saves/Batocera.PLUS/pacman
+
+LIST_OFFLINE=${PACMAN_DIR}/pacman.conf
+LIST_ONLINE=https://raw.githubusercontent.com/AlexxandreFS/Batocera.PLUS/master/pacman.txt
+
+if [ ! -e "${LIST_OFFLINE}" ] || [ ! -L "${HOME}/pacman" ]
+then
+	mkdir -p "${PACMAN_DIR}"
+	rm -rf   "${HOME}/pacman"
+	ln -s    "${PACMAN_DIR}" "${HOME}/pacman"
+
+	if curl -sfL "${LIST_ONLINE}" -o "${LIST_OFFLINE}"
+	then
+		pacman --noconfirm -Syu
+	else
+		exit 1
+	fi
+fi
+
 ACTION=$1
 shift
 
@@ -13,6 +32,8 @@
 	if [ $? -eq 1 ]; then
 		sleep 1
 		pgrep pacman >/dev/null || rm -f /userdata/system/pacman/db/db.lck
+	else
+		yes | LC_ALL=C pacman -Scc
 	fi
 	;;
     "remove")
@@ -24,7 +45,7 @@
 	fi
 	;;
     "list")
-	if test ! -e /userdata/system/pacman/db/sync/batocera.db
+	if test ! -e /userdata/system/pacman/db/sync/batocera.plus.db
 	then
 		pacman --noconfirm -Syu
 	fi
@@ -33,6 +54,12 @@
 
 	;;
     "update")
+	if curl -sfL "${LIST_ONLINE}" -o /tmp/pacman.conf.tmp
+	then
+		mv -f /tmp/pacman.conf.tmp "${LIST_OFFLINE}"
+	else
+		rm -f /tmp/pacman.conf.tmp
+	fi
 	pacman --noconfirm -Syu
 	;;
     *)
