--- batocera-save-overlay.original
+++ batocera-save-overlay.modificado
###
### Aumenta o tamanho padrão do arquivo overlay.
###
@@ -6,7 +6,7 @@
 OVERLAYFILE="/boot/boot/overlay"
 OVERLAYMOUNT="/overlay/saved"
 OVERLAYRAM="/overlay/overlay"
-OVERLAYSIZE=50 # M
+OVERLAYSIZE=128 # M
 
 # fix for winscp running bash and then not sourcing /etc/profile
 PATH=$PATH:/sbin

###
### Monta a partição de forma síncrona e evita que o arquivo overlay seja corrompido.
###
@@ -25,7 +25,7 @@
 
 # the overlay is saved on /boot, make it rw
 echo "Making /boot writable..."
-if ! mount -o remount,rw /boot
+if ! mount -o remount,rw,sync /boot
 then
     exit 1
 fi
@@ -50,7 +50,7 @@
 
 # mount it
 echo "Mounting the overlay file..."
-if ! mount -o rw "${OVERLAYFILE}" "${OVERLAYMOUNT}"
+if ! mount -o rw,sync "${OVERLAYFILE}" "${OVERLAYMOUNT}"
 then
     mount -o remount,ro /boot
     exit 1

###
### Desativa a montagem da unidade como somente leitura se estiver rodando pelo batocera-hd-dition.
###
@@ -74,10 +74,13 @@
 fi
 
 # put /boot in ro back
-echo "Making /boot read only..."
-if ! mount -o remount,ro /boot
+if ! df | grep ^/dev/loop0 | grep /boot$ > /dev/null
 then
-    exit 1
+    echo "Making /boot read only..."
+    if ! mount -o remount,ro /boot
+    then
+        exit 1
+    fi
 fi
 
 echo "Synchronizing..."
