--- winetricks-original
+++ winetricks-modificado
@@ -1,4 +1,32 @@
-#!/bin/sh
+#!/bin/bash
+
+# Batocera.PLUS Start
+
+echo 'Batocera.PLUS'
+
+WINE_BUILD="$(basename "${0}")"
+
+if [ "${WINE_BUILD}" == 'winetricks' ]; then
+    WINE_BUILD='wine-stable'
+else
+    WINE_BUILD="${WINE_BUILD#winetricks-}"
+fi
+
+if [ -z "${WINEPREFIX}" ]; then
+    WINEPREFIX="${HOME}/configs/wine/${WINE_BUILD}"
+    echo 'Warning: It is not recommended to install additional in the standard prefix, some games may stop working.'
+fi
+
+export WINEPREFIX
+export PATH="${PATH}:/opt/Wine/${WINE_BUILD}/bin"
+export CORE="${WINE_BUILD}"
+
+unset WINE_BUILD
+
+XDG_CACHE_HOME='/userdata/downloads'
+
+# Batocera.PLUS Stop
+
 # shellcheck disable=SC2030,SC2031
 # SC2030: Modification of WINE is local (to subshell caused by (..) group).
 # SC2031: WINE was modified in a subshell. That change might be lost
@@ -3418,6 +3446,10 @@
 
 winetricks_latest_version_check()
 {
+    # Batocera.PLUS Start
+    return 0
+    # Batocera.PLUS Stop
+
     if [ "${WINETRICKS_LATEST_VERSION_CHECK}" = 'disabled' ] || [ -f "${WINETRICKS_CONFIG}/disable-latest-version-check" ] ; then
         w_info "winetricks latest version check update disabled"
         return
@@ -3527,8 +3559,10 @@
             WINETRICKS_MENU_HEIGHT=500
             WINETRICKS_MENU_WIDTH=1010
         else
-            echo "Invalid argument for --gui"
-            echo "Valid options are 'zenity' and 'kdialog'"
+            # Batocera.PLUS Start
+            # echo "Invalid argument for --gui"
+            # echo "Valid options are 'zenity' and 'kdialog'"
+            # Batocera.PLUS Stop
             exit 1
         fi
     elif [ "${XDG_CURRENT_DESKTOP}" = "KDE" ] && test -x "$(command -v kdialog 2>/dev/null)"; then
@@ -5313,11 +5347,13 @@
         W_COMMONFILES="$(w_pathconv -u "${W_COMMONFILES_WIN}")"
 
         # 64-bit prefixes still have plenty of issues:
-        case ${LANG} in
-            ru*) w_warn "Вы используете 64-битный WINEPREFIX. Важно: многие ветки устанавливают только 32-битные версии пакетов. Если у вас возникли проблемы, пожалуйста, проверьте еще раз на чистом 32-битном WINEPREFIX до отправки отчета об ошибке." ;;
-            pt*) w_warn "Você está usando um WINEPREFIX de 64-bit. Observe que muitos casos instalam apenas versões de pacotes de 32-bit. Se você encontrar problemas, teste novamente em um WINEPREFIX limpo de 32-bit antes de relatar um bug." ;;
-            *) w_warn "You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug." ;;
-        esac
+        # Batocera.PLUS Start
+        # case ${LANG} in
+        #     ru*) w_warn "Вы используете 64-битный WINEPREFIX. Важно: многие ветки устанавливают только 32-битные версии пакетов. Если у вас возникли проблемы, пожалуйста, проверьте еще раз на чистом 32-битном WINEPREFIX до отправки отчета об ошибке." ;;
+        #     pt*) w_warn "Você está usando um WINEPREFIX de 64-bit. Observe que muitos casos instalam apenas versões de pacotes de 32-bit. Se você encontrar problemas, teste novamente em um WINEPREFIX limpo de 32-bit antes de relatar um bug." ;;
+        #     *) w_warn "You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug." ;;
+        # esac
+        # Batocera.PLUS Stop
     else
         WINE64="false"
         WINE_ARCH="${WINE}"
@@ -5454,9 +5490,11 @@
     fi
 
     # Running Wine as root is (generally) bad, mmkay?
-    if [ "$(id -u)" = 0 ]; then
-        w_warn "Running Wine/winetricks as root is highly discouraged. See https://wiki.winehq.org/FAQ#Should_I_run_Wine_as_root.3F"
-    fi
+    # Batocera.PLUS Start
+    # if [ "$(id -u)" = 0 ]; then
+    #     w_warn "Running Wine/winetricks as root is highly discouraged. See https://wiki.winehq.org/FAQ#Should_I_run_Wine_as_root.3F"
+    # fi
+    # Batocera.PLUS Stop
 
     # Ephemeral files for this run
     WINETRICKS_WORKDIR="${W_TMP_EARLY}/w.${LOGNAME}.$$"
@@ -5653,9 +5691,11 @@
 
     # If WINE is < 5.0, warn user:
     # 5.0 doesn't do what I thought it would
-    if w_wine_version_in ,4.99 ; then
-        w_warn "Your version of wine ${_wine_version_stripped} is no longer supported upstream. You should upgrade to 5.x"
-    fi
+    # Batocera.PLUS Start
+    # if w_wine_version_in ,4.99 ; then
+    #     w_warn "Your version of wine ${_wine_version_stripped} is no longer supported upstream. You should upgrade to 5.x"
+    # fi
+    # Batocera.PLUS Stop
 
     winetricks_set_wineprefix "$1"
 
