--- winetricks.original
+++ winetricks.modificado
@@ -1,4 +1,29 @@
 #!/bin/sh
+
+################################################################################
+
+### Batocera.PLUS
+
+WINE_BUILD="$(basename "${0}")"
+
+if [ "${WINE_BUILD}" == 'winetricks' ]; then
+    WINE_BUILD='wine-stable'
+else
+    WINE_BUILD="${WINE_BUILD#winetricks-}"
+fi
+
+if [ ! "${WINEPREFIX}" ]; then
+    WINEPREFIX="${HOME}/configs/wine/${WINE_BUILD}"
+fi
+
+export WINEPREFIX
+export PATH=":${PATH}:/opt/Wine/${WINE_BUILD}/bin"
+export CORE="${WINE_BUILD}"
+
+unset WINE_BUILD
+
+################################################################################
+
 # shellcheck disable=SC2030,SC2031
 # SC2030: Modification of WINE is local (to subshell caused by (..) group).
 # SC2031: WINE was modified in a subshell. That change might be lost
@@ -5309,9 +5334,9 @@
     fi
 
     # Running Wine as root is (generally) bad, mmkay?
-    if [ "$(id -u)" = 0 ]; then
-        w_warn "Running Wine/winetricks as root is highly discouraged. See https://wiki.winehq.org/FAQ#Should_I_run_Wine_as_root.3F"
-    fi
+    #if [ "$(id -u)" = 0 ]; then
+    #    w_warn "Running Wine/winetricks as root is highly discouraged. See https://wiki.winehq.org/FAQ#Should_I_run_Wine_as_root.3F"
+    #fi
 
     # Ephemeral files for this run
     WINETRICKS_WORKDIR="${W_TMP_EARLY}/w.${LOGNAME}.$$"
@@ -5502,9 +5527,9 @@
 
     # If WINE is < 4.0, warn user:
     # 4.0 doesn't do what I thought it would
-    if w_wine_version_in ,3.99 ; then
-        w_warn "Your version of wine ${_wine_version_stripped} is no longer supported upstream. You should upgrade to 4.x"
-    fi
+    #if w_wine_version_in ,3.99 ; then
+    #    w_warn "Your version of wine ${_wine_version_stripped} is no longer supported upstream. You should upgrade to 4.x"
+    #fi
 
     winetricks_set_wineprefix "$1"
 
