diff --git a/libobs/CMakeLists.txt b/libobs/CMakeLists.txt
index de9b414..449652c 100644
--- a/libobs/CMakeLists.txt
+++ b/libobs/CMakeLists.txt
@@ -28,7 +28,9 @@ if(UNIX)
 		else()
 			set(USE_XINPUT "0")
 		endif()
-		find_package(PulseAudio)
+		if(NOT DISABLE_PULSEAUDIO)
+			find_package(PulseAudio)
+		endif()
 		if (NOT "${PULSEAUDIO_LIBRARY}" STREQUAL "")
 			message(STATUS "Found PulseAudio - Audio Monitor enabled")
 			set(HAVE_PULSEAUDIO "1")
