--- batocera.conf
+++ batocera.conf
#
# Por padrão desativa o botão X para abrir o Kodi.
#
@@ -16,7 +16,7 @@
 ## Start kodi at launch (0,1)
 kodi.atstartup=0
 ## set x button shortcut (0,1)
-kodi.xbutton=1
+kodi.xbutton=0
 
 ## Kodi can wait for a network component before starting
 ## waithost is the ip or hostname that must answer to a ping to validate the availability
#
# Desativa atualizações.
#
@@ -127,7 +127,7 @@
 
 # ------------ G - UPDATES ------------ #
 ## Automatically check for updates at start (0,1)
-updates.enabled=1
+updates.enabled=0
 # default : stable ; set to beta to get the next version currently being tested. set to unstable at your own risk to get the development version.
 updates.type=stable

#
# Ativa o fullboot do GameCube.
# 
@@ -219,6 +219,11 @@
 # Enables fast booting (skip PS2 BIOS when booting an ISO ; some games relies on BIOS to get start-up parameters and thus may not work correctly)
 # set fullboot to 0 to disable BIOS when booting an ISO
 #ps2.fullboot=1
+
+## GameCube
+# Enables fast booting (skip GameCube BIOS when booting an ISO
+# uncoment this line to disable BIOS when booting an ISO
+#gamecube.fullboot=0

+# Enables the use of real gamecube pads on dolphin via MyFlash USB adaptor
+# Just remove the (#) from begining of line above
+#use.gamecube.pads=1
 
 # scrapper
 # Comma seperated order to prefer images, s=snapshot, b=boxart, f=fanart, a=banner, l=logo, 3b=3D boxart
