#
# Adiciona alguns filtros de imagem personalizados.
#
--- rendering-defaults.yml.original
+++ rendering-defaults.yml.modificado
@@ -106,3 +106,32 @@
   shader: crt/fakelottes
 x68000:
   shader: crt/fakelottes
+# Batocera.PLUS
+arcade:
+  shader: crt/fakelottes
+channelf:
+  shader: crt/fakelottes
+fba_libretro:
+  shader: crt/fakelottes
+hbmame:
+  shader: crt/fakelottes
+n64dd:
+  shader: crt/fakelottes
+neogeomini:
+  shader: crt/fakelottes
+neogeox:
+  shader: crt/fakelottes
+nesmini:
+  shader: crt/fakelottes
+pspminis:
+  shader: crt/fakelottes
+psxmini:
+  shader: crt/fakelottes
+pc98:
+  shader: crt/fakelottes
+snesmini:
+  shader: crt/fakelottes
+snescd:
+  shader: crt/fakelottes
+thomson:
+  shader: crt/fakelottes
