#
# Adiciona alguns filtros de imagem personalizados.
#
--- rendering-defaults.yml.original
+++ rendering-defaults.yml.modificado
@@ -106,3 +106,32 @@
   shader: anti-aliasing/advanced-aa
 x68000:
   shader: anti-aliasing/advanced-aa
+# Batocera.PLUS
+arcade:
+  shader: scalehq/2xScaleHQ
+channelf:
+  shader: anti-aliasing/advanced-aa
+fba_libretro:
+  shader: scalehq/2xScaleHQ
+hbmame:
+  shader: scalehq/2xScaleHQ
+n64dd:
+  shader: anti-aliasing/advanced-aa
+neogeomini:
+  shader: scalehq/2xScaleHQ
+neogeox:
+  shader: scalehq/2xScaleHQ
+nesmini:
+  shader: anti-aliasing/advanced-aa
+pspminis:
+  shader: anti-aliasing/advanced-aa
+psxmini:
+  shader: scalehq/2xScaleHQ
+pc98:
+  shader: anti-aliasing/advanced-aa
+snesmini:
+  shader: anti-aliasing/advanced-aa
+snescd:
+  shader: anti-aliasing/advanced-aa
+thomson:
+  shader: anti-aliasing/advanced-aa
