--- es_features.cfg.original
+++ es_features.cfg.modificado
@@ -77,8 +77,224 @@
   <emulator name="melonds" features="videomode" />
   <emulator name="daphne" features="videomode, ratio" />
   <emulator name="flycast" features="videomode, ratio" />
+
+  <!-- BATOCERA.PLUS START -->
+
+  <emulator name="cemu" features="">
+    <cores>
+      <core name="cemu" features="" />
+    </cores>
+    <feature name="OTIMIZATIONS" value="otimizations">
+      <choice name="nvidia" value="nvidia" />
+      <choice name="amd" value="amd" />
+      <choice name="intel" value="intel" />
+    </feature>
+    <feature name="RENDER" value="render">
+      <choice name="opengl" value="opengl" />
+      <choice name="vulkan" value="vulkan" />
+    </feature>
+    <feature name="ESYNC" value="esync">
+      <choice name="on" value="on" />
+      <choice name="off" value="off" />
+    </feature>
+    <feature name="DXVK" value="dxvk">
+      <choice name="on" value="on" />
+      <choice name="off" value="off" />
+    </feature>
+    <!-- FSYNC (Futura implementação)
+    <feature name="FSYNC" value="fsync">
+      <choice name="on" value="on" />
+      <choice name="off" value="off" />
+    </feature> -->
+    <feature name="SHOW FPS" value="showfps">
+      <choice name="top left" value="topleft" />
+      <choice name="top center" value="topcenter" />
+      <choice name="top right" value="topright" />
+      <choice name="botton left" value="bottonleft" />
+      <choice name="botton center" value="bottoncenter" />
+      <choice name="botton right" value="bottonright" />
+    </feature>
+  </emulator>
+
+  <emulator name="model2" features="videomode, ratio">
+    <cores>
+      <core name="singlecore" features="" />
+      <core name="multicore" features="" />
+    </cores>
+    <feature name="ANTIALIASING" value="antialiasing">
+      <choice name="off" value="off" />
+      <choice name="on" value="on" />
+    </feature>
+    <feature name="ANISOTROPIC FILTER" value="anisotropicfilter">
+      <choice name="off" value="off" />
+      <choice name="bilinear" value="bilinear" />
+      <choice name="trilinear" value="trilinear" />
+    </feature>
+    <feature name="VSYNC" value="vsync">
+      <choice name="off" value="off" />
+      <choice name="on" value="on" />
+    </feature>
+    <feature name="DRAW CROSS" value="drawcross">
+      <choice name="off" value="off" />
+      <choice name="on" value="on" />
+    </feature>
+    <feature name="RUMBLE" value="rumble">
+      <choice name="off" value="off" />
+      <choice name="on" value="on" />
+    </feature>
+  </emulator>
+
+  <emulator name="model3" features="videomode, ratio, pixel_perfect">
+    <cores>
+      <core name="singlecore" features="" />
+      <core name="multicore" features="" />
+    </cores>
+    <feature name="MULTI TEXTURE" value="multitexture">
+      <choice name="off" value="off" />
+      <choice name="on" value="on" />
+    </feature>
+    <feature name="MULTI THREAD" value="multithread">
+      <choice name="off" value="off" />
+      <choice name="on" value="on" />
+    </feature>
+    <feature name="QUAD RENDERING" value="quadrendering">
+      <choice name="off" value="off" />
+      <choice name="on" value="on" />
+    </feature>
+    <feature name="VSYNC" value="vsync">
+      <choice name="off" value="off" />
+      <choice name="on" value="on" />
+    </feature>
+    <feature name="NEW 3D ENGINE" value="new3dengine">
+      <choice name="off" value="off" />
+      <choice name="on" value="on" />
+    </feature>
+    <feature name="THROTTLE" value="throttle">
+      <choice name="off" value="off" />
+      <choice name="on" value="on" />
+    </feature>
+    <feature name="POWERPC FREQUENCY" value="powerpcfrequency">
+      <choice name="10" value="10" />
+      <choice name="20" value="20" />
+      <choice name="30" value="30" />
+      <choice name="40" value="40" />
+      <choice name="50" value="50" />
+      <choice name="60" value="60" />
+      <choice name="70" value="70" />
+      <choice name="80" value="80" />
+      <choice name="90" value="90" />
+      <choice name="100" value="100" />
+    </feature>
+    <feature name="RUMBLE" value="rumble">
+      <choice name="off" value="off" />
+      <choice name="on" value="on" />
+    </feature>
+    <feature name="EMULATE NET" value="emulatenet">
+      <choice name="off" value="off" />
+      <choice name="on" value="on" />
+    </feature>
+  </emulator>
+
+  <emulator name="raine" features="smooth, shaders, ratio">
+    <cores>
+      <core name="raine2020" features="" />
+      <core name="raine2015" features="" />
+    </cores>
+    <feature name="BIOS" value="bios">
+      <choice name="cdz" value="cdz" />
+      <choice name="front-load" value="front-load" />
+      <choice name="top-load" value="top-load" />
+      <choice name="unibios" value="unibios" />
+    </feature>
+  </emulator>
+
+  <emulator name="wine" features="">
+    <cores>
+      <core name="wine-stable" features="" />
+      <core name="wine-staging" features="" />
+      <core name="wine-lutris" features="" />
+    </cores>
+    <feature name="GAME EXECUTABLE" value="arch">
+      <choice name="32-bits" value="32" />
+      <choice name="64-bits" value="64" />
+    </feature>
+    <feature name="DXVK" value="dxvk">
+      <choice name="on" value="on" />
+      <choice name="on+fps" value="fps" />
+      <choice name="off" value="off" />
+    </feature>
+    <feature name="MOUSE POINTER" value="mouse_pointer">
+      <choice name="on" value="on" />
+      <choice name="off" value="off" />
+    </feature>
+    <feature name="EMULATE JOYPAD" value="emulate_joypad">
+      <choice name="on" value="on" />
+      <choice name="off" value="off" />
+    </feature>
+    <feature name="SCRIPT BATOCERA.PLUS" value="script">
+      <choice name="on" value="on" />
+      <choice name="off" value="off" />
+    </feature>
+  </emulator>
+
+  <emulator name="proton" features="">
+    <cores>
+      <core name="proton-valve" features="" />
+      <core name="proton-ge-custom" features="" />
+      <core name="proton-tkg" features="" />
+    </cores>
+    <feature name="GAME EXECUTABLE" value="arch">
+      <choice name="32-bits" value="32" />
+      <choice name="64-bits" value="64" />
+    </feature>
+    <feature name="DXVK" value="dxvk">
+      <choice name="on" value="on" />
+      <choice name="on+fps" value="fps" />
+      <choice name="off" value="off" />
+    </feature>
+    <feature name="MOUSE POINTER" value="mouse_pointer">
+      <choice name="on" value="on" />
+      <choice name="off" value="off" />
+    </feature>
+    <feature name="EMULATE JOYPAD" value="emulate_joypad">
+      <choice name="on" value="on" />
+      <choice name="off" value="off" />
+    </feature>
+    <feature name="SCRIPT BATOCERA.PLUS" value="script">
+      <choice name="on" value="on" />
+      <choice name="off" value="off" />
+    </feature>
+  </emulator>
+
   <emulator name="libretro" features="videomode, ratio, smooth, shaders, pixel_perfect, decoration, latency_reduction, game_translation">
     <cores>
+      <core name="bsnes_hd"        features="netplay, rewind, autosave" />
+      <core name="catsfc"          features="netplay, rewind, autosave" />
+      <core name="dolphin"         features="netplay, rewind, autosave" />
+      <core name="duckstation"     features="netplay, rewind, autosave" />
+      <core name="fbalpha2012"     features="netplay, rewind, autosave" />
+      <core name="flycast2019"     features="netplay, rewind, autosave" />
+      <core name="freechaf"        features="netplay, rewind, autosave" />
+      <core name="gearsystem"      features="netplay, rewind, autosave" />
+ <!-- <core name="gpsp"            features="netplay, rewind, autosave" /> -->
+      <core name="hbmame"          features="netplay, rewind, autosave" />
+      <core name="mame0174"        features="netplay, rewind, autosave" />
+      <core name="mame0200"        features="netplay, rewind, autosave" />
+      <core name="mame078"         features="netplay, rewind, autosave" />
+      <core name="mednafen_pce"    features="netplay, rewind, autosave" />
+      <core name="mednafen_psx_sw" features="netplay, rewind, autosave" />
+      <core name="melonds"         features="netplay, rewind, autosave" />
+      <core name="meteor"          features="netplay, rewind, autosave" />
+      <core name="nekop2"          features="netplay, rewind, autosave" />
+      <core name="ppsspp"          features="netplay, rewind, autosave" />
+      <core name="quicknes"        features="netplay, rewind, autosave" />
+      <core name="race"            features="netplay, rewind, autosave" />
+      <core name="redream"         features="netplay, rewind, autosave" />
+      <core name="smsplus"         features="netplay, rewind, autosave" />
+      <core name="yabause"         features="netplay, rewind, autosave" />
+
+  <!-- BATOCERA.PLUS END -->
+
       <core name="gw" features="" />
       <core name="freeintv" features="" />
       <core name="pcfx" features="netplay, rewind, autosave" />
@@ -169,7 +385,6 @@
   </emulator>
   <emulator name="vice" features="videomode, ratio" />
   <emulator name="fsuae" features="videomode, ratio" />
-  <emulator name="cemu" features="videomode" />
   <emulator name="citra" features="videomode, ratio, rewind, screen_layout" />
   <emulator name="moonlight" features="videomode, ratio" />
   <emulator name="rpcs3" features="videomode" />
